src/main/java/com/spikes2212/command/genericsubsystem/smartmotorcontrollersubsystem/SparkMaxGenericSubsystem.java

```diff
@@ -0,0 +1,114 @@
+package com.spikes2212.command.genericsubsystem.smartmotorcontrollersubsystem;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkMaxPIDController;
+import com.spikes2212.command.DashboardedSubsystem;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.control.TrapezoidProfileSettings;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+
+import java.util.List;
+
+/**
+ * A subsystem which consists of a {@link CANSparkMax} motor controller that runs PID loops and additional
+ * {@link CANSparkMax} motor controllers that follow it.
+ *
+ * @author Yoel Perman Brilliant
+ * @see DashboardedSubsystem
+ * @see SmartMotorControllerSubsystem
+ */
+public class SparkMaxGenericSubsystem extends DashboardedSubsystem implements SmartMotorControllerSubsystem {
+
+    /**
+     * The slot on the {@link CANSparkMax} on which the trapezoid profiling configurations are saved.
+     */
+    private static final int TRAPEZOID_SLOT_ID = 0;
+
+    /**
+     * The {@link CANSparkMax} which runs the loops.
+     */
+    protected final CANSparkMax master;
+
+    /**
+     * Additional {@link CANSparkMax}s that follow the master.
+     */
+    protected final List<CANSparkMax> slaves;
+
+    /**
+     * Constructs a new instance of {@link CTRESmartMotorControllerGenericSubsystem}.
+     *
+     * @param namespaceName the name of the subsystem's namespace
+     * @param master        the motor controller which runs the loops
+     * @param slaves        additional motor controllers that follow the master
+     */
+    public SparkMaxGenericSubsystem(String namespaceName, CANSparkMax master, CANSparkMax... slaves) {
+        super(namespaceName);
+        this.master = master;
+        this.slaves = List.of(slaves);
+        this.slaves.forEach(s -> s.follow(master));
+    }
+
+    /**
+     * Adds any data or commands to the {@link Shuffleboard}.
```

Comment:
Technically it puts it on the network tables?

Reply:
ditto